src/relax/transform/fuse_tir.cc

```diff
@@ -438,9 +440,35 @@ class FusedTIRConstructor : public ExprVisitor {
     auto it = func_info_.expr2buffers.find(body);
     ICHECK(it != func_info_.expr2buffers.end())
         << "Fail to detect output buffers for function body";
+
     const Array<tir::Buffer>& buffers = (*it).second;
+
+    // map of input buffers to indices (helpful for detecting in-place inputs)
+    std::unordered_map<tir::Buffer, Integer, ObjectPtrHash, ObjectPtrEqual> buffer_to_idx;
+    std::unordered_map<tir::Var, Integer, ObjectPtrHash, ObjectPtrEqual> input_to_idx;
+    for (size_t i = 0; i < func_info_.params.size(); i++) {
+      input_to_idx[func_info_.params[i]] = Integer(i);
+    }
+    for (auto kv : func_info_.buffer_map) {
+      if (input_to_idx.count(kv.first)) {
+        buffer_to_idx[kv.second] = input_to_idx[kv.first];
+      }
+    }
+
+    // numbered separately because the number of output *vars* might differ from the
+    // number of outputs if there are in-place inputs
+    int out_idx = 0;
     for (size_t i = 0; i < buffers.size(); ++i) {
-      tir::Var param = tir::Var("p_output" + std::to_string(i), PrimType(DataType::Handle()));
+      // Do not add output vars for in-place inputs
+      // (i.e., already listed in the buffer map. This would result
+      // in duplicates in the buffer map otherwise)
+      if (buffer_to_idx.count(buffers[i])) {
```

Comment:
Nitpick: Using `.count` followed by `operator[]` requires two lookups into the map.  This should be avoided, by using `.find` instead.

```c++
if (auto it = buffer_to_idx.count(buffers[i]); it != buffer_to_idx.end()) {
    inplace_indices_.push_back(it.second);
}
```

Reply:
Good advice